src/JsonSchema/ResourceMetadataTrait.php

```diff
@@ -0,0 +1,101 @@
+<?php
+
+/*
+ * This file is part of the API Platform project.
+ *
+ * (c) Kévin Dunglas <dunglas@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace ApiPlatform\JsonSchema;
+
+use ApiPlatform\Metadata\CollectionOperationInterface;
+use ApiPlatform\Metadata\Exception\OperationNotFoundException;
+use ApiPlatform\Metadata\HttpOperation;
+use ApiPlatform\Metadata\Operation;
+use ApiPlatform\Metadata\Resource\ResourceMetadataCollection;
+use ApiPlatform\Metadata\Util\ResourceClassInfoTrait;
+
+/**
+ * @internal
+ */
+trait ResourceMetadataTrait
+{
+    use ResourceClassInfoTrait;
+
+    private function findOutputClass(string $className, string $type, Operation $operation, ?array $serializerContext): ?string
+    {
+        $inputOrOutput = ['class' => $className];
+        $inputOrOutput = Schema::TYPE_OUTPUT === $type ? ($operation->getOutput() ?? $inputOrOutput) : ($operation->getInput() ?? $inputOrOutput);
+        $forceSubschema = $serializerContext[SchemaFactory::FORCE_SUBSCHEMA] ?? false;
+
+        return $forceSubschema ? ($inputOrOutput['class'] ?? $inputOrOutput->class ?? $operation->getClass()) : ($inputOrOutput['class'] ?? $inputOrOutput->class ?? null);
+    }
+
+    private function findOperation(string $className, string $type, ?Operation $operation, ?array $serializerContext): Operation
+    {
+        if (null === $operation) {
+            $resourceMetadataCollection = $this->resourceMetadataFactory->create($className);
```

Comment:
```suggestion
            $resourceMetadataCollection = $this->resourceMetadataFactory->create($className);
```

`resourceMetadataFactory` can be null sometimes, I wonder why phpstan didn't catch that

Reply:
TBH, I seem to remember having a similar thought and originally just left it with the assumption _"If it isn't broken, don't fix it"_